.github/workflows/3rd-party-check.yml

```diff
@@ -0,0 +1,83 @@
+name: 3rd Party Dependency Check
+
+on:
+  pull_request:
+    branches:
+      - staging
+    paths:
+      - 'package.json'
+      - 'package-lock.json'
+
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  check-3rd-party:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Ensures all history is fetched
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      - name: Check for 3rd Party Dependencies
+        id: check_dependencies
+        run: |
+          # Install required tools
+          sudo apt-get install -y jq
+
+          # Fetch the base branch
+          git fetch origin ${{ github.base_ref }}
+          git checkout ${{ github.base_ref }}
+          git checkout -
+
+          # Initialize a flag
+          echo "DEPENDENCY_CHANGE=false" >> $GITHUB_ENV
+
+          # Compare package.json
+          echo "Checking package.json for changes..."
+          if git diff --name-only origin/${{ github.base_ref }}...HEAD | grep 'package.json'; then
+            echo "Changes detected in package.json"
+            echo "DEPENDENCY_CHANGE=true" >> $GITHUB_ENV
+          fi
+
+          # Compare package-lock.json
+          echo "Checking package-lock.json for changes..."
+          if git diff --name-only origin/${{ github.base_ref }}...HEAD | grep 'package-lock.json'; then
+            echo "Changes detected in package-lock.json"
+            echo "DEPENDENCY_CHANGE=true" >> $GITHUB_ENV
+          fi
+
+      - name: Post Comment
+        if: ${{ env.DEPENDENCY_CHANGE == 'true' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+            PR_NUMBER=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+            echo "PR_NUMBER: $PR_NUMBER"
+  
+            COMMENT_BODY='### New 3rd party dependency justification\n'
+            COMMENT_BODY+='If you are looking to implement a new library as part of your pull request, you **must** fill out the information required in the form. Type your answers between the two `|  |` at the end of each line. _Please delete if not relevant._\n\n'
```

Comment:
In the bit that says "Please delete if not relevant," let's update this to say something like: "If this is just an update to an already existing package, you can ignore this".